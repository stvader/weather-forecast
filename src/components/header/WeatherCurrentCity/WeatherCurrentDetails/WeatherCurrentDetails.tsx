import React from 'react';

import { Table } from 'antd';
import { ColumnProps } from 'antd/es/table';

import { ITableColumn, ITableData } from '.';

interface IWeatherCurrentDetailsView {
  columns: ColumnProps<ITableColumn>[];
  data: ITableData;
}

const WeatherCurrentDetails: React.FC<IWeatherCurrentDetailsView> = ({
  data,
  columns,
}: IWeatherCurrentDetailsView) => (
  <Table columns={columns} dataSource={data} showHeader pagination={false} bordered size="small" />
);

export default WeatherCurrentDetails;
