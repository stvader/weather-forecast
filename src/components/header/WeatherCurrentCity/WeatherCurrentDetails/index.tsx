import React, { useContext, ReactElement } from 'react';

import { ColumnProps } from 'antd/es/table';

import { WeatherCurrentContext } from '../WeatherCurrentCityContext';
import WeatherCurrentDetails from './WeatherCurrentDetails';
import WindCityWeather from '../../../WindCityWeather';

export interface ITableColumn {
  key: string;
  wind: ReactElement;
  cloudiness: string;
  pressure: string;
  humidity: string;
  sunrise: string;
  sunset: string;
  coords: string;
}

export type ITableData = ITableColumn[];

const columns: ColumnProps<ITableColumn>[] = [
  {
    title: 'Wind',
    dataIndex: 'wind',
    key: 'wind',
  },
  {
    title: 'Cloudiness',
    dataIndex: 'cloudiness',
    key: 'cloudiness',
  },
  {
    title: 'Pressure',
    dataIndex: 'pressure',
    key: 'pressure',
  },
  {
    title: 'Humidity',
    dataIndex: 'humidity',
    key: 'humidity',
  },
  {
    title: 'Sunrise',
    dataIndex: 'sunrise',
    key: 'sunrise',
  },
  {
    title: 'Sunset',
    dataIndex: 'sunset',
    key: 'sunset',
  },
  {
    title: 'Geo coords',
    dataIndex: 'coords',
    key: 'coords',
  },
];

const WeatherCurrentDetailsContainer: React.FC<any> = () => {
  const {
    wind: { direction, speed },
    cloudiness,
    pressure,
    humidity,
    sunrise,
    sunset,
    coord: { lon, lat },
  } = useContext(WeatherCurrentContext);

  const data: ITableData = [
    {
      key: '1',
      wind: <WindCityWeather direction={direction} speed={speed} />,
      cloudiness,
      pressure: `${pressure} hpa`,
      humidity: `${humidity}%`,
      sunrise,
      sunset,
      coords: `[${lon}, ${lat}]`,
    },
  ];

  return <WeatherCurrentDetails data={data} columns={columns} />;
};

export default WeatherCurrentDetailsContainer;
