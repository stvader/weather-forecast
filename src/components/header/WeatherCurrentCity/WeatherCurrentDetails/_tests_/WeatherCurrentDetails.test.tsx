import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Table } from 'antd';
import WeatherCurrentDetails from '../WeatherCurrentDetails';

configure({ adapter: new Adapter() });

const mockColumns = [
  {
    title: 'Wind',
    dataIndex: 'wind',
    key: 'wind',
  },
  {
    title: 'Cloudiness',
    dataIndex: 'cloudiness',
    key: 'cloudiness',
  },
  {
    title: 'Pressure',
    dataIndex: 'pressure',
    key: 'pressure',
  },
  {
    title: 'Humidity',
    dataIndex: 'humidity',
    key: 'humidity',
  },
  {
    title: 'Sunrise',
    dataIndex: 'sunrise',
    key: 'sunrise',
  },
  {
    title: 'Sunset',
    dataIndex: 'sunset',
    key: 'sunset',
  },
  {
    title: 'Geo coords',
    dataIndex: 'coords',
    key: 'coords',
  },
];

const mockData = [
  {
    key: '1',
    wind: `North, 5 m/h`,
    cloudiness: 'cloudness',
    pressure: `100 hpa`,
    humidity: `98%`,
    sunrise: '9:4',
    sunset: '9:5',
    coords: `[50, 50]`,
  },
];

describe('TemperatureSwitcher', () => {
  it('render TemperatureSwitcher', () => {
    const component = shallow(<WeatherCurrentDetails columns={mockColumns} data={mockData} />);

    const table = component.find(Table);

    expect(component).toMatchSnapshot();
    expect(table).toBeDefined();
    expect(table.prop('columns')).toBe(mockColumns);
    expect(table.prop('dataSource')).toBe(mockData);
    expect(table.prop('showHeader')).toBe(true);
    expect(table.prop('pagination')).toBe(false);
    expect(table.prop('bordered')).toBe(true);
    expect(table.prop('size')).toBe('small');
  });
});
