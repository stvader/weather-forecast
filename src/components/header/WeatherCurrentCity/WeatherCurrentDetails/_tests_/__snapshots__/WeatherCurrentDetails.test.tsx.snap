// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureSwitcher render TemperatureSwitcher 1`] = `ShallowWrapper {}`;
